### PR TITLE
Feature/dex 616 card network icons

### PR DESCRIPTION
**Pull request by EvansPie**
DEX-616

# What this PR does

Shows card network icon when detected
Shows validation icon

_QA_

- [x] I manually tested this with a demo application (simulator)
- [ ] I manually tested this with a demo application (real device)
- [ ] I wrote unit tests for each new util-like function
- [ ] I wrote at least basic e2e tests for the new functionalities or new paths
- [ ] If this PR introduces UI changes, I manually tested this PR on iPhone 8 (iOS 10, 12, 14), iPhone 12, and iPhone 12 Max
- [ ] If this PR modifies the behavior of an API, I did my best to make my new API backward compatible
- [ ] If this PR modifies the API, I attempted to implement the feature with an example

_Documentation_

- [ ] If this PR modifies the API, I updated the API Reference.
- [ ] If this PR adds new options to the API, I wrote some documentation or guide in the online documentation
- [ ] If the PR modifies the API, I wrote some documentation to deprecate it

# After merging

- Make sure a new release has been pushed to Cocoapods trunk

### CODE REVIEW

File: Sources/PrimerSDK/Classes/Data Models/CardNetwork.swift [hunk 1]
```diff
@@ -232,7 +232,36 @@ public enum CardNetwork: String, CaseIterable {
...
     
     public var icon: UIImage? {
-        return nil
+        switch self {
+        case .amex:
+            return UIImage(named: "amex", in: Bundle.primerResources, compatibleWith: nil)
+        case .bancontact:
+            return UIImage(named: "bancontact-icon", in: Bundle.primerResources, compatibleWith: nil)
+        case .diners:
+            return UIImage(named: "genericCard", in: Bundle.primerResources, compatibleWith: nil)
```
Comment (ceriksson) [inline Sources/PrimerSDK/Classes/Data Models/CardNetwork.swift:241]: maybe we should raise a request to have icons for these other ones too. Wdyt?

File: Sources/PrimerSDK/Classes/User Interface/Root/PrimerCardFormViewController.swift [hunk 2]
```diff
@@ -258,15 +258,111 @@ extension PrimerCardFormViewController: CardComponentsManagerDelegate, PrimerTex
...
-        } else if primerTextFieldView is PrimerExpiryDateFieldView, isValid == false {
-            expiryDateContainerView.errorText = "Invalid date"
-        } else if primerTextFieldView is PrimerCVVFieldView, isValid == false {
-            cvvContainerView.errorText = "Invalid CVV"
-        } else if primerTextFieldView is PrimerCardholderNameFieldView, isValid == false {
-            cardholderNameContainerView.errorText = "Invalid name"
+        if isValid == false && !primerTextFieldView.isEmpty {
+            if primerTextFieldView is PrimerCardNumberFieldView {
+                cardNumberContainerView.errorText = "Invalid card number"
+                cardNumberContainerView.rightImage1 = UIImage(
```
Comment (ceriksson) [inline Sources/PrimerSDK/Classes/User Interface/Root/PrimerCardFormViewController.swift:264]: Is there a way we could place these images in constants in a separate file? It becomes quite difficult to determine the logic when everything is all in one place.

File: Sources/PrimerSDK/Classes/Data Models/CardNetwork.swift (hunk 1 above)
Comment (EvansPie) [reply Sources/PrimerSDK/Classes/Data Models/CardNetwork.swift:241]: yeah, I looked for them in Figma and they aren't there.

File: Sources/PrimerSDK/Classes/Extensions & Utilities/PrimerImage.swift [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+//
+//  Created by Evangelos Pittas on 20/10/21.
+//
+
+#if canImport(UIKit)
+
+import UIKit
+
+internal enum PrimerImage {
+    case amexCardIcon, appleIcon, backIcon, bankcontactCardIcon, bankIcon, cameraIcon,
```
Comment (ceriksson) [inline Sources/PrimerSDK/Classes/Extensions & Utilities/PrimerImage.swift:13]: nice!

File: Sources/PrimerSDK/Classes/User Interface/Root/PrimerCardFormViewController.swift (hunk 2 above)
Comment (ceriksson) [reply Sources/PrimerSDK/Classes/User Interface/Root/PrimerCardFormViewController.swift:266]: thx!

**Pull request merged**